tools/pepsi/src/progress_indicator.rs

```diff
@@ -0,0 +1,67 @@
+use std::{borrow::Cow, time::Duration};
+
+use indicatif::{MultiProgress, ProgressBar, ProgressStyle};
+
+#[derive(Clone)]
+pub struct ProgressIndicator {
+    multi_progress: MultiProgress,
+    default_style: ProgressStyle,
+    error_style: ProgressStyle,
+    success_style: ProgressStyle,
+}
+
+impl ProgressIndicator {
+    pub fn new() -> Self {
+        Self {
+            multi_progress: MultiProgress::new(),
+            default_style: ProgressStyle::with_template("{prefix:.bold.dim} {spinner} {wide_msg}")
+                .unwrap()
+                .tick_chars("⠋⠙⠹⠸⠼⠴⠦⠧⠇⠏"),
+            error_style: ProgressStyle::with_template("{prefix:.bold.dim} {wide_msg:.red}")
+                .unwrap(),
+            success_style: ProgressStyle::with_template("{prefix:.bold.dim} {wide_msg:.green}")
+                .unwrap(),
+        }
+    }
+
+    pub fn task(&self, prefix: String) -> Task {
+        let spinner = ProgressBar::new(10)
+            .with_style(self.default_style.clone())
+            .with_prefix(prefix);
+        spinner.enable_steady_tick(Duration::from_millis(100));
```

Comment:
I would be in favor of increasing that tick interval to e.g. 500 ms because we want to spend our CPU resources on executing our tasks instead of butterly smooth animations... 😉

Reply:
I disagree, I'm sure even your Gurke can handle updating some text ten times per second without meaningful impact on the actual tasks.